agents/agents-core/src/commonMain/kotlin/ai/koog/agents/core/agent/SingleRunStrategyWithHistoryCompression.kt

```diff
@@ -0,0 +1,126 @@
+package ai.koog.agents.core.agent
```

Comment:
Let's move this whole file to the `agents-ext` module to the `agent` package, to keep core modules as lean as possible.